haskell-indexer-backend-ghc/src/Language/Haskell/Indexer/Backend/GhcApiSupport.hs

```diff
@@ -287,23 +297,45 @@ generateDocUriDecls globalPkgs refs =
     needsDocUri t = (isNothing . tickSpan) t && isFromCorePackage t
     isFromCorePackage t = (getPackage . tickPkgModule) t `S.member` globalPkgs
 
+-- | Generates doc/uri information for imported modules that are from core
+-- packages. Because those modules are not indexed, add Hackage document URLs
+-- instead.
+generateModuleDocUriDecls ::
+  Set Text -> Set PkgModule -> [ModuleTick] -> [ModuleDocUriDecl]
+generateModuleDocUriDecls globalPkgs indexedModules imports =
+  let mts = nubOrdOn mtPkgModule $ imports
+   in mapMaybe generate mts
+  where
+    generate :: ModuleTick -> Maybe ModuleDocUriDecl
+    generate mt =
+      let pm = mtPkgModule mt
+       in if needsDocUri pm
+            then Just $ ModuleDocUriDecl mt (hackageSrcUrlForModule pm)
+            else Nothing
+    needsDocUri pm = not (pm `S.member` indexedModules) && isFromCorePackage pm
```

Comment:
From my limited understanding, ti seems like this code is structured a little differently than generateDocUriDecls since it takes the indexedModules parameter explicitly.  Can you help me understand why the difference was needed?